src/main/java/newitem/NewItemController.java

```diff
@@ -1,18 +1,33 @@
 package newitem;
 
-import entities.Product;
 
 public class NewItemController {
 
-    private final NewItemInteractor newItemInteractor;
+    private final NewItemInputBoundary inputBoundary;
 
-    public NewItemController() {
-        this.newItemInteractor = new NewItemInteractor();
+
+    public NewItemController(NewItemInputBoundary inputBoundary) {
+        this.inputBoundary = inputBoundary;
     }
 
-    //returns Product for testing purposes but maybe that's what the user will be given upon execution of the request
-    public Product newItem(String name, long UPC, int price) {
-        return this.newItemInteractor.addNewItem(name, UPC, price);
-        //TODO: return logic for presenter
+    public NewItemResponseModel newItem(String name, String sUPC, String sPrice) {
+
+        try {
+            if (sUPC.length() != 12) {
+                throw new Exception();
+            }
+            long upc = Long.parseLong(sUPC);
+            int price = Integer.parseInt(sPrice);
+            if (price < 0) {
+                throw new Exception();
+            }
+            NewItemRequestModel request = new NewItemRequestModel(name, upc, price);
+            return inputBoundary.newItem(request);
+
+        } catch (Exception e) {
```

Comment:
this catches all exceptions, it would be good to create a custom exception to throw and catch to avoid any unwanted error messages being thrown